ambry-clustermap/src/main/java/com.github.ambry.clustermap/StaticClusterAgentsFactory.java

```diff
@@ -79,9 +81,13 @@ public ClusterParticipant getClusterParticipant() throws IOException {
         private final List<PartitionStateChangeListener> listeners = new LinkedList<>();
 
         @Override
-        public void participate(List<AmbryHealthReport> ambryHealthReports) {
+        public void participate(List<AmbryHealthReport> ambryHealthReports, DataNodeId currentNode) {
+          Set<String> partitionsOnNode = getClusterMap().getReplicaIds(currentNode)
+              .stream()
+              .map(r -> r.getPartitionId().toPathString())
+              .collect(Collectors.toSet());
           for (PartitionStateChangeListener listener : listeners) {
-            for (String partitionName : partitionLayout.getAllPartitionNames()) {
+            for (String partitionName : partitionsOnNode) {
```

Comment:
What are the implications of calling this callback on the partitions not present on this node?